content/blogposts/2023/copilot-vs-cody.md

```diff
@@ -0,0 +1,229 @@
+description: "Does your AI coding assistant really understand your code? We tested GitHub Copilot vs Cody on a production codebase to see the difference context makes."
+tags: [blog]
+slug: 'copilot-vs-cody-context-driven-development'
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/copilot-vs-cody-blog.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/copilot-vs-cody-blog.png
+---
+
+Code AI assistants like GitHub Copilot and [Cody](https://cody.dev) promise developers increased productivity, better workflows, and an improved developer experience. There is a lot of hype around these tools and many promises made, but how good are these tools at actually delivering on these promises?
+
+In this post, I want to put two of the most popular coding assistants head to head and see if they can improve a developer's experience. We'll put the code AI assistants through various real-world scenarios and see if they make the developer's life easier or harder.
+
+## Setup
+
+For this blog post, I wanted to use a production-grade application to test how the two code AI assistants handle scenarios, from teaching me about a codebase to helping me get set up with it, adding new functionality, debugging, and more.
+
+I will be using [Visual Studio Code](https://code.visualstudio.com/), and I have two versions installed locally: the latest stable version of Visual Studio Code and an Insiders version. On each of these versions, I have either [Cody](https://cody.dev) or Copilot installed as the only extension. I will be using the same prompts for each assistant and will provide thoughts and commentary on their output. The codebase we'll be using for these scenarios is for a web application called [Video Tap](https://videotap.com) that allows users to repurpose their long-form video-based content into various written and short-form video formats. Let's dive in!
+
+## Scenario: Learning about a codebase
+
+For our first scenario, let's place ourselves in the shoes of a new developer onboarding onto a codebase. Without a code AI assistant, the developer would likely first look at the `readme.md` file to see if it gives a good overview or perhaps reach out to the maintainer of the codebase for a walkthrough.
+
+In my personal experience, the `readme.md` files of projects are often hit or miss. They may include an amazing overview and instructions, or they may be completely blank and everything in between. With a code AI assistant that supports chat functionality, we can ask it to tell us what a codebase does. Let's try it with Cody and Copilot to see which one does better.
+
+Our query will be: **"What does this application do?"**
+
+![Explain what a codebase does](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/explain-what-codebase-does.png)
```

Comment:
Can we give this a clearer label so the user doesn't have to consult the paragraph to determine which is Cody vs Copilot?